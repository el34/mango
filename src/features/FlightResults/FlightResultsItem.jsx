import React from "react";
import { Row, Col, List, Typography, Timeline, Divider, Tag } from "antd";

const { Title, Text } = Typography;

const formatDayOptions = { weekday: "short", month: "long", day: "numeric" };
const formatTimeOptions = { minutes: "numeric", hours: "numeric" };

export const formatDateFromTimestamp = (ts, type) => {
  return type === "day"
    ? new Date(ts * 1000).toLocaleDateString("en-US", formatDayOptions)
    : new Date(ts * 1000).toLocaleTimeString("en-US", formatTimeOptions);
};

export const FlightResultsItem = ({ item }) => {
  return (
    <List.Item key={item.id} style={{ display: "flex", width: "100%" }}>
      <Row style={{ display: "flex", width: "100%" }}>
        <Col span={24}>
          <Row
            style={{
              flexDirection: "column",
              marginBottom: "1.2rem",
            }}
          >
            <Title level={5} style={{ marginBottom: "0" }}>
              {formatDateFromTimestamp(
                item.routesToDestionation[0].dTimeUTC,
                "day"
              )}
            </Title>
            <Text secondary>
              {" "}
              {item.cityFrom} to {item.cityTo}
            </Text>
          </Row>
          <Row>
            <Timeline style={{ marginTop: "12px" }}>
              <Timeline.Item>
                <Text strong>
                  {formatDateFromTimestamp(
                    item.routesToDestionation[0].dTimeUTC,
                    false
                  )}
                  {", "}
                  {item.cityFrom}
                </Text>
                <Text>{`, ${item.countryFrom.name}`}</Text>
                <Text secondary> {`(${item.cityCodeFrom})`}</Text>
              </Timeline.Item>
              <Timeline.Item>
                <Text strong>
                  {formatDateFromTimestamp(
                    item.routesToDestionation[
                      item.routesToDestionation.length - 1
                    ].aTimeUTC,
                    false
                  )}
                  {", "}
                  {item.cityTo}
                </Text>
                <Text>{`, ${item.countryTo.name}`}</Text>
                <Text secondary> {`(${item.cityCodeTo})`}</Text>
              </Timeline.Item>
            </Timeline>
          </Row>
          <Row>
            <Tag color="volcano">{item.fly_duration}</Tag>
            <Tag color="blue">{`nights in destination: ${item.nightsInDest}`}</Tag>
          </Row>
          <Divider
            style={{ marginTop: "0", marginBottom: "0" }}
            orientation="right"
          >
            <>
              <Title level={3} style={{ margin: "0", display: "block" }}>
                {item.price} EUR
              </Title>
              <Text strong>{`Seats: ${item.availability.seats}`}</Text>
            </>
          </Divider>
          <Row
            style={{
              flexDirection: "column",
              marginBottom: "1.2rem",
            }}
          >
            <Title level={5} style={{ marginBottom: "0" }}>
              {formatDateFromTimestamp(
                item.routesFromDestionation[0].dTimeUTC,
                "day"
              )}
            </Title>
            <Text secondary>
              {" "}
              {item.cityTo} to {item.cityFrom}
            </Text>
          </Row>
          <Row>
            <Timeline style={{ marginBottom: "12px" }}>
              <Timeline.Item>
                <Text strong>
                  {formatDateFromTimestamp(
                    item.routesFromDestionation[0].dTimeUTC,
                    false
                  )}
                  {", "}
                  {item.cityTo}
                </Text>
                <Text>{`, ${item.countryTo.name}`}</Text>
                <Text secondary>{` (${item.cityCodeTo})`}</Text>
              </Timeline.Item>
              <Timeline.Item>
                <Text strong>
                  {formatDateFromTimestamp(
                    item.routesFromDestionation[
                      item.routesFromDestionation.length - 1
                    ].aTimeUTC,
                    false
                  )}
                  {", "}
                  {item.cityFrom}
                </Text>
                <Text>{`, ${item.countryFrom.name}`}</Text>
                <Text secondary>{` (${item.cityCodeFrom})`}</Text>
              </Timeline.Item>
            </Timeline>
          </Row>
          <Row>
            <Tag color="volcano">{item.return_duration}</Tag>
          </Row>
        </Col>
      </Row>
    </List.Item>
  );
};
