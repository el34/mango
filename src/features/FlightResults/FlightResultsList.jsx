import React from 'react'

export const FlightResultsList = () => {
  return (
    <div>FlightResultsList</div>
  )
}
