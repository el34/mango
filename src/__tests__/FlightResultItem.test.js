import { FlightResultsItem } from "../features/FlightResults/FlightResultsItem";
import { create } from "react-test-renderer";
import { flightResultItemMock as item } from "./flightResultItemMock";

test("FlightResultsItem renders correctly with flight", () => {
  const tree = create(<FlightResultsItem item={item} />).toJSON();
  expect(tree).toMatchSnapshot();
});
