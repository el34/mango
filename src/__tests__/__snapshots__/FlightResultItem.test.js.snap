// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightResultsItem renders correctly with flight 1`] = `
<li
  className="ant-list-item"
  style={
    Object {
      "display": "flex",
      "width": "100%",
    }
  }
>
  <div
    className="ant-row"
    style={
      Object {
        "display": "flex",
        "width": "100%",
      }
    }
  >
    <div
      className="ant-col ant-col-24"
      style={Object {}}
    >
      <div
        className="ant-row"
        style={
          Object {
            "flexDirection": "column",
            "marginBottom": "1.2rem",
          }
        }
      >
        <h5
          className="ant-typography"
          onClick={null}
          style={
            Object {
              "WebkitLineClamp": undefined,
              "marginBottom": "0",
            }
          }
        >
          Fri, September 30
        </h5>
        <span
          className="ant-typography"
          onClick={null}
          secondary="true"
          style={
            Object {
              "WebkitLineClamp": undefined,
            }
          }
        >
           Prague to Barcelona
        </span>
      </div>
      <div
        className="ant-row"
        style={Object {}}
      >
        <ul
          className="ant-timeline"
          style={
            Object {
              "marginTop": "12px",
            }
          }
        >
          <li
            className="ant-timeline-item"
          >
            <div
              className="ant-timeline-item-tail"
            />
            <div
              className="ant-timeline-item-head ant-timeline-item-head-blue"
              style={
                Object {
                  "borderColor": undefined,
                  "color": undefined,
                }
              }
            />
            <div
              className="ant-timeline-item-content"
            >
              <span
                className="ant-typography"
                onClick={null}
                style={
                  Object {
                    "WebkitLineClamp": undefined,
                  }
                }
              >
                <strong>
                  3:55:00 PM, Prague
                </strong>
              </span>
              <span
                className="ant-typography"
                onClick={null}
                style={
                  Object {
                    "WebkitLineClamp": undefined,
                  }
                }
              >
                , Czechia
              </span>
              <span
                className="ant-typography"
                onClick={null}
                secondary="true"
                style={
                  Object {
                    "WebkitLineClamp": undefined,
                  }
                }
              >
                 
                (PRG)
              </span>
            </div>
          </li>
          <li
            className="ant-timeline-item ant-timeline-item-last"
          >
            <div
              className="ant-timeline-item-tail"
            />
            <div
              className="ant-timeline-item-head ant-timeline-item-head-blue"
              style={
                Object {
                  "borderColor": undefined,
                  "color": undefined,
                }
              }
            />
            <div
              className="ant-timeline-item-content"
            >
              <span
                className="ant-typography"
                onClick={null}
                style={
                  Object {
                    "WebkitLineClamp": undefined,
                  }
                }
              >
                <strong>
                  6:15:00 PM, Barcelona
                </strong>
              </span>
              <span
                className="ant-typography"
                onClick={null}
                style={
                  Object {
                    "WebkitLineClamp": undefined,
                  }
                }
              >
                , Spain
              </span>
              <span
                className="ant-typography"
                onClick={null}
                secondary="true"
                style={
                  Object {
                    "WebkitLineClamp": undefined,
                  }
                }
              >
                 (BCN)
              </span>
            </div>
          </li>
        </ul>
      </div>
      <div
        className="ant-row"
        style={Object {}}
      >
        <span
          className="ant-tag ant-tag-volcano"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        >
          2h 20m
        </span>
        <span
          className="ant-tag ant-tag-blue"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        >
          nights in destination: 1
        </span>
      </div>
      <div
        className="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-right"
        role="separator"
        style={
          Object {
            "marginBottom": "0",
            "marginTop": "0",
          }
        }
      >
        <span
          className="ant-divider-inner-text"
          style={Object {}}
        >
          <h3
            className="ant-typography"
            onClick={null}
            style={
              Object {
                "WebkitLineClamp": undefined,
                "display": "block",
                "margin": "0",
              }
            }
          >
            317 EUR
          </h3>
          <span
            className="ant-typography"
            onClick={null}
            style={
              Object {
                "WebkitLineClamp": undefined,
              }
            }
          >
            <strong>
              Seats: 2
            </strong>
          </span>
        </span>
      </div>
      <div
        className="ant-row"
        style={
          Object {
            "flexDirection": "column",
            "marginBottom": "1.2rem",
          }
        }
      >
        <h5
          className="ant-typography"
          onClick={null}
          style={
            Object {
              "WebkitLineClamp": undefined,
              "marginBottom": "0",
            }
          }
        >
          Sat, October 1
        </h5>
        <span
          className="ant-typography"
          onClick={null}
          secondary="true"
          style={
            Object {
              "WebkitLineClamp": undefined,
            }
          }
        >
           Barcelona to Prague
        </span>
      </div>
      <div
        className="ant-row"
        style={Object {}}
      >
        <ul
          className="ant-timeline"
          style={
            Object {
              "marginBottom": "12px",
            }
          }
        >
          <li
            className="ant-timeline-item"
          >
            <div
              className="ant-timeline-item-tail"
            />
            <div
              className="ant-timeline-item-head ant-timeline-item-head-blue"
              style={
                Object {
                  "borderColor": undefined,
                  "color": undefined,
                }
              }
            />
            <div
              className="ant-timeline-item-content"
            >
              <span
                className="ant-typography"
                onClick={null}
                style={
                  Object {
                    "WebkitLineClamp": undefined,
                  }
                }
              >
                <strong>
                  8:40:00 AM, Barcelona
                </strong>
              </span>
              <span
                className="ant-typography"
                onClick={null}
                style={
                  Object {
                    "WebkitLineClamp": undefined,
                  }
                }
              >
                , Spain
              </span>
              <span
                className="ant-typography"
                onClick={null}
                secondary="true"
                style={
                  Object {
                    "WebkitLineClamp": undefined,
                  }
                }
              >
                 (BCN)
              </span>
            </div>
          </li>
          <li
            className="ant-timeline-item ant-timeline-item-last"
          >
            <div
              className="ant-timeline-item-tail"
            />
            <div
              className="ant-timeline-item-head ant-timeline-item-head-blue"
              style={
                Object {
                  "borderColor": undefined,
                  "color": undefined,
                }
              }
            />
            <div
              className="ant-timeline-item-content"
            >
              <span
                className="ant-typography"
                onClick={null}
                style={
                  Object {
                    "WebkitLineClamp": undefined,
                  }
                }
              >
                <strong>
                  9:55:00 PM, Prague
                </strong>
              </span>
              <span
                className="ant-typography"
                onClick={null}
                style={
                  Object {
                    "WebkitLineClamp": undefined,
                  }
                }
              >
                , Czechia
              </span>
              <span
                className="ant-typography"
                onClick={null}
                secondary="true"
                style={
                  Object {
                    "WebkitLineClamp": undefined,
                  }
                }
              >
                 (PRG)
              </span>
            </div>
          </li>
        </ul>
      </div>
      <div
        className="ant-row"
        style={Object {}}
      >
        <span
          className="ant-tag ant-tag-volcano"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        >
          13h 15m
        </span>
      </div>
    </div>
  </div>
</li>
`;
